### PR TITLE
PODAAC-4327 - Migrate NSIDC ISO changes

### PR DESCRIPTION
**Pull request by jocelynmallon**
Migrated NSIDC ISO related changes from old/archived repo.

### CODE REVIEW

File: src/main/java/gov/nasa/cumulus/metadata/aggregator/UMMGranuleFile.java [hunk 1]
```diff
@@ -515,6 +406,132 @@ private JSONObject exportSpatial() {
...
+            double north = 0, south = 0, east = 0, west = 0;
+            if(granule !=null && granule instanceof  gov.nasa.cumulus.metadata.aggregator.UMMGranule) {
+                east = ((UMMGranule) granule).getBbxEasternLongitude() != null ?
+                        ((UMMGranule) granule).getBbxEasternLongitude() : 0;
+                west = ((UMMGranule) granule).getBbxWesternLongitude() != null?
+                        ((UMMGranule) granule).getBbxWesternLongitude() : 0;
+                north = ((UMMGranule) granule).getBbxNorthernLatitude() != null?
+                        ((UMMGranule) granule).getBbxNorthernLatitude() : 0;
+                south = ((UMMGranule) granule).getBbxSouthernLatitude() != null?
+                        ((UMMGranule) granule).getBbxSouthernLatitude() : 0;
```
Comment (michaeljb) [inline src/main/java/gov/nasa/cumulus/metadata/aggregator/UMMGranuleFile.java:427]: This section is new--looks like it's just making sure there are north/south/east/west values present? Setting to 0 if they're null?

**Review by michaeljb — COMMENTED** [review]
I've commented a couple places where I spotted differences, but it looks like nothing was lost from our original PR. Thanks!

File: src/main/java/gov/nasa/cumulus/metadata/aggregator/UMMGranuleFile.java (hunk 1 above)
Comment (jocelynmallon) [reply src/main/java/gov/nasa/cumulus/metadata/aggregator/UMMGranuleFile.java:427]: Right, if I remember correctly, this was related to some granules we encountered with bad spatial metadata, but the UMM-C still required spatial extent, but there were also just a lot of little differences between the internal codebase we were working from, and the previous GitHub repository.

File: src/main/java/gov/nasa/cumulus/metadata/aggregator/MetadataAggregatorLambda.java [hunk 2]
```diff
@@ -128,18 +128,22 @@ public String PerformFunction(String input, Context context) throws Exception {
...
-		if (iso != null) {
-            mtfe = new MetadataFilesToEcho(true);
-            mtfe.setDatasetValues(collectionName, collectionVersion, rangeIs360, boundingBox);
+        boolean isIsoFile = (iso != null);
+
+        mtfe = new MetadataFilesToEcho(isIsoFile);
+        mtfe.setDatasetValues(collectionName, collectionVersion, rangeIs360, boundingBox);
+        if (granules != null && granules.size() > 0) {
+            mtfe.setGranuleFileSizeAndChecksum(granules);
+        }
```
[the remark this answers is not in the document]
Comment (jocelynmallon) [reply src/main/java/gov/nasa/cumulus/metadata/aggregator/MetadataAggregatorLambda.java:137]: I'm not 100% sure about this, @yenes56 did most of the Cumulus 11 work, so he'd be the best one to ask. It was in our existing internal repo, but wasn't in the old GitHub repo, so I just grabbed/copied the lines over.

**Review by skorper — APPROVED** [review]

**Pull request merged**